Tribler/community/tunnel/tunnel_community.py

```diff
@@ -1216,22 +1214,35 @@ def crypto_out(self, circuit_id, content, is_data=False):
 
     def crypto_in(self, circuit_id, content, is_data=False):
         circuit = self.circuits.get(circuit_id, None)
-        if circuit and len(circuit.hops) > 0:
-            # Remove all the encryption layers
-            for hop in self.circuits[circuit_id].hops:
-                content = self.crypto.decrypt_str(
-                    content, hop.session_keys[ORIGINATOR], hop.session_keys[ORIGINATOR_SALT])
-            if circuit and is_data and circuit.ctype in [CIRCUIT_TYPE_RENDEZVOUS, CIRCUIT_TYPE_RP]:
-                direction = int(circuit.ctype != CIRCUIT_TYPE_RP)
-                direction_salt = direction + 2
-                content = self.crypto.decrypt_str(
-                    content, circuit.hs_session_keys[direction], circuit.hs_session_keys[direction_salt])
-            return content
+        if circuit:
+            if len(circuit.hops) > 0:
+                # Remove all the encryption layers
+                layer = 0
+                for hop in self.circuits[circuit_id].hops:
+                    layer += 1
+                    try:
+                        content = self.crypto.decrypt_str(
+                            content, hop.session_keys[ORIGINATOR], hop.session_keys[ORIGINATOR_SALT])
+                    except InvalidTag as e:
```

Comment:
@whirm 
Just FYI. The times I saw an InvalidTag error, it was caused by one candidate that was added as hop two or more times after each other in a single circuit (which should not happen anyway), then the InvalidTag error occured while peeling of the encryption layers. 
So if the error occurs again, it might be useful to have a look which hops are participating in the circuit.